hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/HoodieRealtimeRecordReader.java

```diff
@@ -76,9 +76,14 @@
   // Depending on outputfile size and memory provided, choose true to avoid OOM for large file
   // size + small memory
   public static final String COMPACTION_LAZY_BLOCK_READ_ENABLED_PROP =
-      "compaction.lazy.block" + ".read.enabled";
+      "compaction.lazy.block.read.enabled";
   public static final String DEFAULT_COMPACTION_LAZY_BLOCK_READ_ENABLED = "true";
 
+  // Property to set the max memory for hdfs inputstream buffer size
+  public static final String MAX_HDFS_STREAM_BUFFER_SIZE_PROP = "hoodie.memory.hdfs.buffer.max.size";
+  // Setting this to lower value of 1 MB since no control over how many RecordReaders will be started in a mapper
+  public static final int DEFAULT_MAX_HDFS_STREAM_BUFFER_SIZE = 1 * 1024 * 1024; // 1 MB
```

Comment:
@vinothchandar Kept this lower since don't know memory settings for mapper. WDYT ?